src/ray/core_worker/transport/direct_task_transport.h

```diff
@@ -71,34 +75,44 @@ class CoreWorkerDirectTaskSubmitter {
   /// \param[in] task_spec The task to schedule.
   Status SubmitTask(TaskSpecification task_spec);
 
-  /// Callback for when the raylet grants us a worker lease. The worker is returned
-  /// to the raylet once it finishes its task and either the lease term has
-  /// expired, or there is no more work it can take on.
-  ///
-  /// \param[in] addr The (addr, port) pair identifying the worker.
-  void HandleWorkerLeaseGranted(const WorkerAddress addr);
-
  private:
   /// Schedule more work onto an idle worker or return it back to the raylet if
   /// no more tasks are queued for submission. If an error was encountered
   /// processing the worker, we don't attempt to re-use the worker.
   void OnWorkerIdle(const WorkerAddress &addr, bool was_error);
 
   /// Request a new worker from the raylet if no such requests are currently in
-  /// flight.
-  void RequestNewWorkerIfNeeded(const TaskSpecification &resource_spec)
+  /// flight and there are tasks queued. If a raylet address is provided, then
+  /// the worker should be requested from the raylet at that address. Else, the
+  /// worker should be requested from the local raylet.
+  void RequestNewWorkerIfNeeded(const TaskSpecification &resource_spec,
+                                const rpc::Address *raylet_address = nullptr)
       EXCLUSIVE_LOCKS_REQUIRED(mu_);
 
+  /// Callback for when the raylet grants us a worker lease. The worker is returned
+  /// to the raylet once it finishes its task and either the lease term has
+  /// expired, or there is no more work it can take on. The worker should be
```

Comment:
This seems not very accurate since we haven't implemented lease expire yet? Could you instead add a TODO here on this part?